src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectIssueResource.java

```diff
@@ -0,0 +1,231 @@
+		
+		RepositoryEntity repositoryEntity = getRepositoryEntity();
+		RepositoryApi repositoryApi = getRepositoryApi(repositoryEntity);
+		RepositoryModel repository = repositoryApi.getRepositoryModel();
+		
+		Map<String, Object> parameters = getBaseParameters();		
+		parameters.put("repository", repository);
+		
+		List<Locale> locales = Collections.list(request.getLocales());
+		return display(templateEngine.process("courses/assignments/group-issue-edit.ftl", locales, parameters));
+	}
+
+	@POST
+	@Transactional
+	@Path("/issues/create")
+	public Response createIssue(@Context HttpServletRequest request,
+			@FormParam("title") String title,
+			@FormParam("description") String description,
+			@FormParam("assignee") String assigneeNetID) throws IOException{
+		
+		Issue issue = new Issue();
+		
+		issue.setTitle(title);
+		issue.setDescription(description);
+		issue.setOpen(true);
+		
+		User assignee = users.findByNetId(assigneeNetID);
+		checkCollaborator(assignee);
+		
+		issue.setAssignee(assignee);
+		issue.setTimestamp(new Date());
+		issue.setRepository(getRepositoryEntity());
+		
+		issueBackend.createIssue(getRepositoryApi(getRepositoryEntity()), issue);
+		
+		return redirect(issue.getURI().toString());
+	}
+	@GET
+	@Transactional
+	@Path("/issue/{issueId}/edit")
+	public Response editIssue(@Context HttpServletRequest request, 
+			@PathParam("issueId") long issueId) throws IOException {
+
+		RepositoryEntity repositoryEntity = getRepositoryEntity();
+		RepositoryApi repositoryApi = getRepositoryApi(repositoryEntity);
+		RepositoryModel repository = repositoryApi.getRepositoryModel();
+		
+		Issue issue = issues.findIssueById(getRepositoryEntity(), issueId).get(0);
+		
+		Map<String, Object> parameters = getBaseParameters();		
+		parameters.put("repository", repository);		
+		parameters.put("issue", issue);
+		
+		List<Locale> locales = Collections.list(request.getLocales());
+		return display(templateEngine.process("courses/assignments/group-issue-edit.ftl", locales, parameters));
+	}
+
+	@GET
+	@Transactional
+	@Path("/issue/{issueId}")
+	public Response viewIssue(@Context HttpServletRequest request, 
+			@PathParam("issueId") long issueId) throws IOException {
+
+		RepositoryEntity repositoryEntity = getRepositoryEntity();
+		RepositoryApi repositoryApi = getRepositoryApi(repositoryEntity);
+		RepositoryModel repository = repositoryApi.getRepositoryModel();
+		
+		Issue issue = issues.findIssueById(getRepositoryEntity(), issueId).get(0);
+		
+		Map<String, Object> parameters = getBaseParameters();		
+		parameters.put("repository", repository);		
+		parameters.put("issue", issue);
+		
+		List<Locale> locales = Collections.list(request.getLocales());
+		return display(templateEngine.process("courses/assignments/group-issue-view.ftl", locales, parameters));
+	}
+
+	@POST
+	@Transactional
+	@Path("/issue/{issueId}/edit")
+	public Response updateIssue(@Context HttpServletRequest request, 
```

Comment:
You can also add `@Context` parameters as fields to your resource class. (They don't require another annotation such as `@Inject`, because they are injected by JAX-RS (and CDI if you would use that) and not Guice). Passing the HttpServletRequest as field rather than method parameter makes it easier to call this methods from tests / other parts of the code.

*No need to change this now*